index.html

```diff
@@ -10442,47 +10572,72 @@ <h2>User prompts</h2>
 
  <dl>
   <dt>"<code>text</code>"
-  <dd>The <a>current user prompt</a>’s
+  <dd>The <a>current user prompt</a>&apos;s
    <a data-lt="user prompt message">message</a>.
  </dl>
 
-<p>To <dfn>handle any user prompts</dfn>
- given <var>session</var>, <a>remote end</a> must:
+<p>To <dfn class=export>get the prompt handler</dfn> <var>type</var>:
 
-<ol>
-<li><p>
-If there is no <a>current user prompt</a>,
-abort these steps and return <a>success</a>.
+<ol class=algorithm>
+ <li><p>If the <a>user prompt handler</a> is null,
+ let <var>handlers</var> be an empty map. Otherwise
+ let <var>handlers</var> be <a>user prompt handler</a>.
 
-<li><p>
-Perform the following substeps
-based on the <a>session</a>’s <a>user prompt handler</a>:
+ <li><p>If <var>handlers</var> contains <var>type</var>
+ return <var>handlers</var>[<var>type</var>].
 
-<dl class=switch>
-<dt><a>dismiss state</a>
-<dd><p><a>Dismiss</a> the <a>current user prompt</a>.
+ <li><p>If <var>type</var> is "<code>beforeUnload</code>", return a
+   <a>prompt handler configuration</a> with [=prompt
+   handler configuration/handler=] "<code>accept</code>"
+   and [=prompt handler configuration/notify=] false.
 
-<dt><a>accept state</a>
-<dd><p><a>Accept</a> the <a>current user prompt</a>.
+ <li><p>If <var>handlers</var> contains "<code>default</code>"
+ return <var>handlers</var>["<code>default</code>"].
 
-<dt><a>dismiss and notify state</a>
-<dd>
-<ol>
-<li><p><a>Dismiss</a> the <a>current user prompt</a>.
-<li><p>Return an <a>annotated unexpected alert open error</a>.
+ <li><p>Return a <a>prompt handler
+   configuration</a> with [=prompt handler
+   configuration/handler=] "<code>dismiss</code>" and [=prompt handler
+   configuration/notify=] true.
 </ol>
 
-<dt><a>accept and notify state</a>
-<dd>
-<ol>
-<li><p><a>Accept</a> the <a>current user prompt</a>.
-<li><p>Return an <a>annotated unexpected alert open error</a>.
-</ol>
+<p>To <dfn>handle any user prompts</dfn>:
+
+<ol class=algorithm>
+ <li><p>If the <a>current browsing context</a> is not blocked by
+ a dialog return <a>success</a>.
+
+ <li><p>Let <var>type</var> be "<code>default</code>".
+
+ <li><p>If the <a>current user prompt</a> is an alert dialog,
+ set <var>type</var> to "<code>alert</code>". Otherwise, if
+ the <a>current user prompt</a> is a confirm dialog,
+ set <var>type</var> to "<code>confirm</code>". Otherwise, if
+ the <a>current user prompt</a> is a prompt dialog,
+ set <var>type</var> to "<code>prompt</code>".
+
+ <li><p>TODO: other types
```

Comment:
Is adding other types blocked on something?

Reply:
On landing this, mostly :) I've replaced this TODO with a comment about handling print and file prompts. I mostly don't want to scope creep this PR with more additional functionality.